woocommerce/woocommerce
# Remove marketing transient on upgrade to 7.0

### All Submissions:






### Changes proposed in this Pull Request:

This PR removes the transient used to cache recommended marketing extensions when WooCommerce is upgraded to version 7.0.

The marketing page calls the `GET https://woocommerce.com/wp-json/wccom/marketing-tab/1.1/recommendations.json` API to retrieve recommended marketing extensions and caches the value in a transient (`wc_marketing_recommended_plugins`). However, the new marketing UI implemented in #33895 utilizes the `GET https://woocommerce.com/wp-json/wccom/marketing-tab/1.2/recommendations.json` API to fetch the recommended marketing extensions. Note that the version is changed from "1.1" to "1.2" in the URL. The new API has a different data structure; therefore, the new marketing page will fail to load if data from the previous API is stored in the `wc_marketing_recommended_plugins` transient.

When WooCommerce is upgraded, we need to remove that cached data from the transient so it can be repopulated using the new API.

Closes #34655.

### How to test the changes in this Pull Request:

1. Activate a previous version of WC (older than 7.0)
2. Visit the marketing page and make sure the list of recommended marketing extensions is stored in the `wc_marketing_recommended_plugins` transient on your DB.
3. Upgrade to WC 7.0 and confirm that the update function is run and the `wc_marketing_recommended_plugins` is deleted.
4. Go to WC Settings > Advanced > Features and enable the new marketing page
5. Check the marketing page again and confirm that you can see the list of recommended marketing extensions

### Other information:




### FOR PR REVIEWER ONLY:
